processing/src/main/java/org/apache/druid/segment/VectorColumnProcessorFactory.java

```diff
@@ -83,6 +83,8 @@ T makeMultiValueDimensionProcessor(
    */
   T makeLongProcessor(ColumnCapabilities capabilities, VectorValueSelector selector);
 
+  T makeArrayProcessor(ColumnCapabilities capabilities, VectorObjectSelector selector);
```

Comment:
the `ColumnProcessors` methods that call this always have `ColumnCapabilities` though via the selector factories, and all of the other methods have the same signature. Given that `ColumnCapabilities` is more rich than `TypeSignature` I think its nicer to leave it as is. For the vector engine, null `ColumnCapabilities` means a column didn't exist in the segment. For now at least I want to be consistent instead of changing the signature, and a number of the other make..processor methods are using stuff that only exists on the capabilities such as checking for multi-value strings, checking capabilities.hasNulls to know if they should check isNull on the selectors, etc.